### ISSUE TITLE
can't insert properly in DATE and TIMESTAMP type

### ISSUE THREAD

**Issue by silverist** [bizstation/transactd#26]
I tried to insert DATE and TIMESTAMP type values by using time format string like 'YYYY-MM-DD' and 'YYYY-MM-DD hh:mm:ss', but the values are invalid like '0000-00-00 00:00:00'.
How can I these type values properly?

**Comment by bizstation** [comment]
Please tell us followings:
- Result of `SHOW CREATE TABLE` SQL command.
- The part of source code which sets values.
- The part of source code which gets and compares values.

**Comment by silverist** [comment]
They are as follows.
"ss_time" column data does not become "2015-10-01".
"addtime" column data becomes "0000-00-00 00:00:00", and "updatetime" column too.

■SHOW CREATE TABLE result
```
CREATE TABLE `lines_pl_ob` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `symbol` varchar(20) NOT NULL,
  `ss_time` date NOT NULL,
  `type` int(11) NOT NULL DEFAULT '0',
  `status` int(11) NOT NULL DEFAULT '0',
  `value` double NOT NULL DEFAULT '0',
  `note` varchar(50) DEFAULT NULL,
  `addtime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updatetime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=111 DEFAULT CHARSET=utf8
```
■The part of source code which sets values
```C++
	database* db = database::create();
	static const _TCHAR* uri = _T("tdap://root@localhost/ilte_line?dbfile=transactd_schema&pwd=r4nk0r4nk0");
	db->open(uri, TYPE_SCHEMA_BDF, TD_OPEN_NORMAL);
	if (db->stat() == 0)
	{
		//テーブルを開く
		table* tb = db->openTable(_T("lines_pl_ob"));

		tb->beginBulkInsert(BULKBUFSIZE);
		TCHAR* note = new TCHAR[100];
		std::setlocale(LC_ALL, "japanese");

		int after_length;

		for (auto pll : pl_line_list)
		{
			after_length = MultiByteToWideChar(CP_UTF8, MB_PRECOMPOSED, pll.note.c_str(), -1, NULL, 0);
			MultiByteToWideChar(CP_UTF8, MB_PRECOMPOSED, pll.note.c_str(), -1, note, after_length);
_tprintf(_T("%s\n"), note);

			tb->clearBuffer();
			tb->setFV(_T("symbol"), pll.symbol.c_str());
			tb->setFV(_T("ss_time"), pll.ss_time.c_str());
			tb->setFV(_T("type"), pll.type);
			tb->setFV(_T("status"), pll.status);
			tb->setFV(_T("value"), pll.value);
			tb->setFV(_T("note"), note);
//			tb->setFV(_T("addtime"), 100);
//			tb->setFV(_T("updatetime"), 100);
			tb->insert();

		}

		tb->commitBulkInsert();
		tb->close();
	}

	// close and release database object
	database::destroy(db);
```
■pl_line_list(insert data, instead of "The part of source code which gets and compares values")
```
symbol : ss_time             : type : status : value  : note
----------------------------------------------------------------
USDJPY : 2015-10-01 21:20:37 : 11   : 0      : 122.00 : 売り厚め
USDJPY : 2015-10-01 21:20:37 : 11   : 0      : 121.50 : 断続的に売り
USDJPY : 2015-10-01 21:20:37 : 11   : 0      : 121.60 : 断続的に売り
USDJPY : 2015-10-01 21:20:37 : 11   : 0      : 121.25 : 断続的に売り
USDJPY : 2015-10-01 21:20:37 : 11   : 0      : 121.30 : 断続的に売り
```

**Comment by bizstation** [comment]
> "ss_time" column data does not become "2015-10-01".

What value have been inserted? NULL-Value like `"0000-00-00"`? Or `NULL`? Or other date?

How did you check inserted values? Did you use `getFV` API in Transactd clients? Did you use `SELECT` SQL? Or other tool?
(I said "The part of source code which gets and compares values", because I want to make sure that data check procedure is not wrong.
I can not clarify that it is in "setFV problem" or "get value problem" with current information.)

**Comment by silverist** [comment]
>How did you check inserted values? Did you use getFV API in Transactd clients? Did you use SELECT SQL? Or other tool?

I check values on "Heidi SQL 9.1.0.4867" which comes with Maria DB 10.0.
(show like the following snap shot.)

■insert data snap shot(on "Heidi SQL 9.1.0.4867" ) 
![_201510271822](https://cloud.githubusercontent.com/assets/10098847/10755245/cd5c59d4-7cdf-11e5-9089-f59a5baaea4c.PNG)

>(I said "The part of source code which gets and compares values", because I want to make sure that data check procedure is not wrong.
I can not clarify that it is in "setFV problem" or "get value problem" with current information.)

I don't use getFV API in the program.(the same as #27)
The above "The part of source code which sets values" is all of the code which "transactd client" use.
In this sense, I think that the answer which "setFV problem" or "get value problem" is probably "setFV problem".

**Comment by bizstation** [comment]
We checked our code and found some bugs.
Maybe, these problems were fixed on Transactd 2.4.5.

Please try to solve the problem with following step:

1. Install Transactd 2.4.5 plugin and clients.
2. Drop `transactd_schema` table on your database. (Note: `transactd_schema` table will be generated automatically on open table if it does not exist.)
3. Retry to insert.

**Comment by silverist** [comment]
I tried the above step on transactd client 2.4.5, but "ss_time" column data does not become "2015-10-01" as the following snap shot.
(According to the page http://www.bizstation.jp/ja/transactd/client/sdk/doc/page_5datatype_map.html, 
"ss_time" column type should be "mydate" on transactd schema?)

![_utf8_heidisql_9_1_201510312354](https://cloud.githubusercontent.com/assets/10098847/10864246/66953802-802b-11e5-8479-4c9d32291798.PNG)


■dumpTableSchema
```
exec codepage   65001
table charset   22
field count     9

name    id
type    AutoIncrement
len     4
charset 1

name    symbol
type    myVarChar
len     61
charset 22

name    ss_time
type    ZString
len     3
charset 1

name    type
type    Integer
len     4
charset 1

name    status
type    Integer
len     4
charset 1

name    value
type    Float
len     8
charset 1

name    note
type    myVarChar
len     151
charset 22

name    addtime
type    myTimeStamp
len     4
charset 1

name    updatetime
type    myTimeStamp
len     4
charset 1
```

**Comment by bizstation** [comment]
> I tried the above step on transactd client 2.4.5

Please check if **SERVER PLUGIN** is version 2.4.5.
Then drop table `transactd_schema` once.
(Transactd server plugin generates `transactd_schema` table automatically.)

**Comment by silverist** [comment]
>Please check if SERVER PLUGIN is version 2.4.5.

I tried them on SERVER PLUGIN version 2.4.4.

I change SERVER PLUGIN version 2.4.5 too, then "ss_time" column becomes "2015-10-01" and "addtime" and "updatetime" become current timestamp by mysql table column setting "DEFAULT CURRENT_TIMESTAMP" as the following snap shot.

Thank you for your support.

![_utf8_heidisql_9_1_201511030700](https://cloud.githubusercontent.com/assets/10098847/10895659/b4a49ce4-81f8-11e5-8077-5315ce6d953d.PNG)

■dumpTableSchema
```
exec codepage   65001
table charset   22
field count     9

name    id
type    AutoIncrement
len     4
charset 1

name    symbol
type    myVarChar
len     61
charset 22

name    ss_time
type    myDate
len     3
charset 1

name    type
type    Integer
len     4
charset 1

name    status
type    Integer
len     4
charset 1

name    value
type    Float
len     8
charset 1

name    note
type    myVarChar
len     151
charset 22

name    addtime
type    myTimeStamp
len     4
charset 1

name    updatetime
type    myTimeStamp
len     4
charset 1
```

**Comment by bizstation** [comment]
Note: Fixed on commit 7a45d9f411b54af594068f433144aed20c3f0a11 .
